File: docs/current/guides/snippets/use-services/test-service-lifecycle-1/main.go
package main

import (
  "context"
  "fmt"
  "os"

  "dagger.io/dagger"
)

func main() {
  ctx := context.Background()

  // create Dagger client
  client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))

  if err != nil {
    panic(err)
  }
  defer client.Close()

  // create Redis service container
  redisSrv := client.Container().
    From("redis").
    WithExposedPort(6379).
    WithExec(nil)

  // create Redis client container
  redisCLI := client.Container().
    From("redis").
    WithServiceBinding("redis-srv", redisSrv).
    WithEntrypoint([]string{"redis-cli", "-h", "redis-srv"})

  // send ping from client to server
  ping := redisCLI.WithExec([]string{"ping"})

  val, err := ping.
    Stdout(ctx)

  if err != nil {
    panic(err)
  }

  fmt.Println(val)
}

File: docs/current/guides/snippets/use-services/test-service-lifecycle-1/main.py
import sys

import anyio
import dagger


async def main():
    # create Dagger client
    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:
        # create Redis service container
        redisSrv = (
            client.container().from_("redis").with_exposed_port(6379).with_exec([])
        )

        # create Redis client container
        redisCli = (
            client.container()
            .from_("redis")
            .with_service_binding("redis-srv", redisSrv)
            .with_entrypoint(["redis-cli", "-h", "redis-srv"])
        )

        # send ping from client to server
        ping = await redisCli.with_exec(["ping"]).stdout()

    print(ping)


anyio.run(main)
